app/ProductPage/page.js

```diff
@@ -0,0 +1,184 @@
+"use client";
+import Image from "next/image";
+import { db } from "@/FirebaseConfig/Firebase";
+import { collection, getDocs } from "firebase/firestore";
+import { BsCartDash, BsPlusSquare } from "react-icons/bs";
+import { AiOutlineDelete, AiOutlineMinusSquare } from "react-icons/ai";
+import { React, useEffect, useState } from "react";
+
+function ProductsPage() {
+  const [cart, setCart] = useState({});
+  const [hidecart, setHideCart] = useState("hidden");
+  const [productList, setProductList] = useState([]);
+  const values = Object.values(cart);
+
+  // INCEARESE ITEM QUANTITY INSIDE CART
+  const quantityIncrement = (item, key) => {
+    setCart({
+      ...cart,
+      [key]: {
+        ...item,
+        quantity: item.quantity + 1,
+      },
+    });
+  };
+  // DECEARESE ITEM QUANTITY INSIDE CART
+  const quantityDecrement = (item, key) => {
+    setCart({
+      ...cart,
+      [key]: {
+        ...item,
+        quantity: item.quantity - 1,
+      },
+    });
+  };
+  // ADD ITEM FROM CART
+  const addItemToCart = (product) => {
+    setCart({ ...cart, [product.id]: { ...product.data, quantity: 1 } });
+  };
+  // REMOVE ITEM FROM CART
+  const removeItemFromCart = (product) => {
+    setCart(
+      Object.assign(
+        {},
+        Object.values(cart).filter((filterTag) => filterTag !== product)
+      )
```

Comment:
[Comment]
I will like you to explain to me on call, how this works

Reply:
ok sir